app/models/chunk.rb

```diff
@@ -14,18 +14,23 @@ class Chunk < ApplicationRecord
   DEFAULT_SIZE = 512
   DEFAULT_OVERLAP = 32
 
-  def self.from_string!(content)
+  def self.from_string!(document, content, with_embedding: true)
     transaction do
       content.each_chunk(DEFAULT_SIZE, DEFAULT_OVERLAP)
              .with_index do |chunk_content, chunk_index|
-        create!(chunk_index:, content: chunk_content)
+        embedding = (EmbeddingService.generate(chunk_content) if with_embedding)
```

Comment:
Probably OK for now, but we need to be careful with introducing external API calls in the context of a transaction since we're holding locks in the db. It also feels like this shouldn't really live in the model layer - could we refactor this so that the embeddings are created somewhere around L42 in generate_new_message_job?